### PR TITLE
Handle SIGINT via async-exit-hook (closes #1378)

### LINKED ISSUE #1378: Try to exit gracefully on SIGINT
### Are you requesting a feature or reporting a bug?
feature

### What is the current behavior?
TestCafe terminates leaving browsers open

### What is the expected behavior?
Try to terminate running tasks gracefully and only after some timeout terminate process if
it didn't work

### PR DESCRIPTION
**Pull request by AndreyBelym**

### CODE REVIEW

File: src/browser/provider/index.js [hunk 1]
```diff
@@ -195,6 +195,14 @@ export default class BrowserProvider {
     }
 
     async closeBrowser (browserId) {
+        var isLocalBrowser    = await this.plugin.isLocalBrowser(browserId);
+        var supportedFeatures = await this.plugin.hasCustomActionForBrowser(browserId);
```
Comment (miherlosev) [inline src/browser/provider/index.js]: `this.plugin.hasCustomActionForBrowser(browserId)` - bad naming convention.

`has` prefix describe that method returns `true` of `false`.

I propose the following variant:
```js
var browserCustomActions = await this.plugin.getCustomActionsForBrowser(browserId)
...
if (isLocalBrowser && !browserCustomActions.closeBrowser) {
```

/cc @AlexanderMoskovkin

File: src/cli/index.js [hunk 2]
```diff
@@ -9,8 +10,50 @@ import log from './log';
 import remotesWizard from './remotes-wizard';
 import createTestCafe from '../';
 
-function exit (code) {
+
+const TERMINATION_TYPES = {
+    sigint:   'sigint',
+    shutdown: 'shutdown'
+};
```
Comment (miherlosev) [inline src/cli/index.js]: On Windows platform is possible to terminate cli by pressing `CTRL+BREAK`.
I propose to support `SIGBREAK` signal.
See https://nodejs.org/api/process.html#process_signal_events

File: src/browser/provider/index.js (hunk 1 above)
Comment (AlexanderMoskovkin) [reply src/browser/provider/index.js]: I agree with this point but it doesn't relate to this PR. And this plugin is already a part of the plugin-host API. If we change it it'll be a breaking change. I think we shouldn't change it in the context of this PR.

Comment (miherlosev) [reply src/browser/provider/index.js]: I've created the separate issue - https://github.com/DevExpress/testcafe/issues/1784

**Comment by AndreyBelym** [comment]
FPR

File: src/browser/provider/index.js [hunk 3]
```diff
@@ -195,6 +195,14 @@ export default class BrowserProvider {
...
+
+        if (isLocalBrowser && !supportedFeatures.hasCloseBrowser) {
+            await this.plugin.closeBrowser(this.windowDescriptors[browserId]);
```
Comment (AlexanderMoskovkin) [inline src/browser/provider/index.js]: In what cases this condition is true?

Comment (AlexanderMoskovkin) [inline src/browser/provider/index.js]: In what cases this condition is true?

**Pull request merged**